### PR TITLE
[FlagGems Operator Development Competition] Add pixel_shuffle operator

### PR DESCRIPTION
**Pull request by yunyiliu**
## Summary

Implement `pixel_shuffle` (sub-pixel convolution) operator.

- **Schema**: `torch.nn.functional.pixel_shuffle(input, upscale_factor) -> Tensor`
- **Category**: layout | **Task #13** | Medium
- Rearranges `(N, C*r², H, W)` → `(N, C, H*r, W*r)` via Triton index mapping
- Inverse of existing `pixel_unshuffle` implementation

## Test Coverage

| Test Type | Coverage |
|-----------|---------|
| upscale_factor=2 (various shapes × 3 dtypes) | tests |
| upscale_factor=3 (various shapes × 3 dtypes) | tests |
| **Total** | **15 tests** (+ 9 auto-skipped for incompatible C/r²) |

Input shapes tested: `(1, 12, 4, 4)`, `(2, 18, 3, 3)`, `(1, 4, 8, 8)`, `(4, 36, 2, 2)`

## Files Changed
- `src/flag_gems/ops/pixel_shuffle.py` - Triton kernel
- `src/flag_gems/ops/__init__.py` - Register exports
- `src/flag_gems/__init__.py` - Register aten dispatch
- `tests/test_unary_pointwise_ops.py` - Accuracy tests

---
**GPU-verified**: All tests passed on NVIDIA RTX PRO 6000 (Blackwell)

### CODE REVIEW

**Comment by bin913** [comment]
Please resolve conflicts

**Comment by yunyiliu** [comment]
Conflicts resolved by merging the latest master. The conflict was in `tests/test_unary_pointwise_ops.py` around the `leaky_relu` test — kept master's updated version and re-inserted the `pixel_shuffle` test after it. PTAL @bin913.

**Comment by bin913** [comment]
> Conflicts resolved by merging the latest master. The conflict was in `tests/test_unary_pointwise_ops.py` around the `leaky_relu` test — kept master's updated version and re-inserted the `pixel_shuffle` test after it. PTAL @bin913.

The tests in test_unary_pointwise_ops.py have been split out. Please refer to other operators to add independent test files.

**Comment by yunyiliu** [comment]
Moved the pixel_shuffle tests into a dedicated `tests/test_pixel_shuffle.py` (modeled on `tests/test_pixel_unshuffle.py`) covering both the functional call and the `.out` variant, and removed the corresponding block from `tests/test_unary_pointwise_ops.py`. PTAL @bin913.

**Comment by yunyiliu** [comment]
Note: the latest `python-op` failure is an infrastructure issue, not a code problem — the runner couldn't `actions/checkout` the repo:

```
fatal: unable to access 'https://github.com/flagos-ai/FlagGems/': Failed to connect to github.com port 443: Connection timed out
```

It happened on all retry attempts before the job ever reached the tests. Could you please re-run the failed job? @bin913

**Comment by bin913** [comment]
> Note: the latest `python-op` failure is an infrastructure issue, not a code problem — the runner couldn't `actions/checkout` the repo:
> 
> ```
> fatal: unable to access 'https://github.com/flagos-ai/FlagGems/': Failed to connect to github.com port 443: Connection timed out
> ```
> 
> It happened on all retry attempts before the job ever reached the tests. Could you please re-run the failed job? @bin913

This branch is out-of-date with the base branch. Please merge master 
389761aa-568e-4e74-9acb-04734b017a57

**Comment by yunyiliu** [comment]
Merged latest master. PTAL @bin913.

**Comment by bin913** [comment]
LGTM

**Review by bin913 — APPROVED** [review]
